cluster-autoscaler/processors/nodegroups/autoprovisioning_nodegroup_manager.go

```diff
@@ -95,3 +95,8 @@ func (p *AutoprovisioningNodeGroupManager) RemoveUnneededNodeGroups(context *con
 	}
 	return nil
 }
+
+// CleanUp cleans up the processor's internal structures.
+func (p *AutoprovisioningNodeGroupManager) CleanUp() error {
```

Comment:
Is there even a point in this (and other CleanUp methods) returning any error? It's not like we can do anything about errors at this stage. And you don't even log them, so it's very easy for someone to implement something that returns an error which won't even be visible in logs.

Reply:
There can be pending tasks on a processor that are flushed in a CleanUp, I think it can be meaningful to be notified about dropping any of them. Also, external clients executing tasks asynchronously may not even notify about any errors until they are closed, which would happen in CleanUp.

Reply:
I fully agree that it may be useful to log some failure in CleanUp. The question is why not leave it to CleanUp implementation to log whatever it thinks is worth logging? Not allowing error makes it more explicit that it's up to the author of specific processor implementation to log if necessary and gives them more control over exactly what is logged, at what level, etc.

Reply:
Fair enough :)

Reply:
@MaciekPytel fixed, everything should be ok now